### PR TITLE
Support Renesas RA GPIO interrupt

### PR DESCRIPTION
**Pull request by thenguyenyf**
- Add support for GPIO interrupt on Renesas RA family
- Enable support of gpio-key for on-board buttons of Renesas RA boards
- Move device node of ioport6, ioport7, ioport8 on r7fa6m4af3cfb to common part of r7fa6m4ax
- Add 1 more whitespace sensitive macro for clang-format: `DT_FOREACH_PROP_ELEM_SEP`

### CODE REVIEW

File: drivers/misc/renesas_ra_external_interrupt/renesas_ra_external_interrupt.c [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+	unsigned int irq;
+};
+
+struct gpio_ra_irq_data {
+	struct gpio_ra_callback callback;
+	struct k_sem irq_sem;
+};
+
+#define _ELC_EVENT_ICU_IRQ(channel) ELC_EVENT_ICU_IRQ##channel
+#define ELC_EVENT_ICU_IRQ(channel)  _ELC_EVENT_ICU_IRQ(channel)
```
Comment (soburi) [inline drivers/misc/renesas_ra_external_interrupt/renesas_ra_external_interrupt.c:45]: You can use `UTIL_CAT` macro.

File: drivers/gpio/gpio_renesas_ra_ioport.c [hunk 2]
```diff
@@ -9,21 +9,56 @@
 #include <zephyr/drivers/gpio.h>
 #include <zephyr/drivers/pinctrl.h>
 #include <zephyr/dt-bindings/gpio/renesas-ra-gpio-ioport.h>
+#include <zephyr/drivers/misc/renesas_ra_external_interrupt/renesas_ra_external_interrupt.h>
 #include <zephyr/drivers/gpio/gpio_utils.h>
-#include <zephyr/irq.h>
 #include <soc.h>
 
+struct gpio_ra_irq_info {
```
Comment (soburi) [inline drivers/gpio/gpio_renesas_ra_ioport.c:16]: I think this implementation is somewhat based on mine, but it wasn't sophisticated enough, so the way this data is held could be made a bit more efficient.

**Review by soburi — CHANGES_REQUESTED** [review]

File: drivers/misc/renesas_ra_external_interrupt/renesas_ra_external_interrupt.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (thenguyenyf) [reply drivers/misc/renesas_ra_external_interrupt/renesas_ra_external_interrupt.c]: I fixed it

**Comment by thenguyenyf** [comment]
Last push to rebase main branch and solve conflicts

File: drivers/gpio/gpio_renesas_ra_ioport.c (hunk 2 above)
Comment (thenguyenyf) [reply drivers/gpio/gpio_renesas_ra_ioport.c:16]: Hi @soburi , could you please give me more about this comment? Which actions should be done for it?

Comment (soburi) [reply drivers/gpio/gpio_renesas_ra_ioport.c:16]: I found my previously considered improving source and looked at it again, but it wasn't all that efficient, so this is fine.
Sorry for stopping.

**Review by soburi — DISMISSED** [review]

**Review by bjarki-andreasen — DISMISSED** [review]
I think this is implemented correctly, but damn, is it complex :D I would add a doc entry somewhere to explain how it works with the `ioports`

**Review by duynguyenxa — DISMISSED** [review]

**Review by soburi — DISMISSED** [review]

**Comment by kartben** [comment]
needs a rebase

**Review by bjarki-andreasen — APPROVED** [review]

**Review by soburi — APPROVED** [review]

**Review by KhiemNguyenT — APPROVED** [review]

**Pull request merged**